iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -32,6 +32,37 @@ static Value gpuAllocationFunction(OpBuilder &builder, Location loc,
   return builder.create<memref::AllocOp>(loc, allocType, dynamicSizes);
 }
 
+//===---------------------------------------------------------------------===//
+// Codegen configuration verifications.
+//===---------------------------------------------------------------------===//
+
+LogicalResult verifyGPUMatmulSimtPassPipeline(
```

Comment:
this is fine for now but I don't see why this should be in the passes.cpp file, not sure why it is done that way for the cpu.

Reply:
Thats where the pipeline is defined. So you add the verifier for the pipeline and the pipeline description side-by-side. (though I do need to move the pipeline description outside the passes file)

Reply:
I agree with Mahesh for the current implementation (keeping them together for simplicity). I'm going to followup with a more involved verifier, which I may break into more functions and cpp files.